test/lint/lint-circular-dependencies.sh

```diff
@@ -20,6 +20,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "txmempool -> validation -> txmempool"
     "wallet/fees -> wallet/wallet -> wallet/fees"
     "wallet/wallet -> wallet/walletdb -> wallet/wallet"
+    "node/coinstats -> validation -> node/coinstats"
```

Comment:
Having to add a circular dependency here is kind of a shame.

Reply:
I'm with ya. This is cached (and may be stale) but I think the only workable alternative is to fold `coinstats` into validation. If you can think of something else I'm happy to make the change.

Reply:
It isn't possible to factor out the part of `node/coinstats` that's used by `validation` to a third module, to prevent the `validation -> node/coinstats` side of the cycle?

Reply:
I don't think we want to roll more into validation, but less.

My only idea here is to have the snapshot loading functionality is maybe not in validation itself, but in a separate implementation unit (which can depend on both).